tests/test_make.py

```diff
@@ -163,24 +163,22 @@ class A:
     assert data == A(1)  # type: ignore[call-arg]
 
 
-def test_list(module: DataclassModule) -> None:
+@pytest.mark.parametrize("annotation_class", (t.List, t.Sequence, t.MutableSequence))
+def test_list(module: DataclassModule, annotation_class: type) -> None:
     """Build a generic list *without* setting a factory on the dataclass."""
-
-    @module.dataclass
-    class A:
-        y: t.List[int]
+    klass = type("A", (object,), {"__annotations__": {"y": annotation_class[int]}})
```

Comment:
(applies to multiple locations)

https://docs.python.org/3.10/library/functions.html#classmethod
> `def f(cls, arg1, arg2, ...): ...`

https://www.python.org/dev/peps/pep-0008/#designing-for-inheritance
> 'cls' is the preferred spelling for any variable or argument which is known to be a class

```suggestion
    cls = type("A", (object,), {"__annotations__": {"y": annotation_class[int]}})
```